common/logbook-date.ts

```diff
@@ -0,0 +1,59 @@
+class DateValidator {
+    // Validates that the input string is a valid date formatted as "mm/dd/yyyy"
+    static isValidDate = (dateString: string): boolean => {
+        // First check for the pattern
+        if (!/^\d{4}\/\d{1,2}\/\d{1,2}$/.test(dateString)) {
+            return false;
+        }
+
+        // Parse the date parts to integers
+        const parts = dateString.split('/');
+        const day = parseInt(parts[2], 10);
+        const month = parseInt(parts[1], 10);
+        const year = parseInt(parts[0], 10);
+
+        // Check the ranges of month and year
+        if (year < 1000 || year > 3000 || month === 0 || month > 12) {
```

Comment:
Not gonna lie I chuckled a bit on the year range 😄 wouldnt be the check for year better to be something like `year < thisYear+1 || year > thisYear-1` ? I dont think we will log hw older than 1 year, but now that i'm thinking about it -> do we use this only for logging? or do we use it for some kind of statistics mambo jumbo or something? 🤔 in that case my proposed year range woudlnt have to work? sorry im kinda thinking "out loud" here now 😄 but still i think it would be nice to somehow solve that `> 3000` 
![image](https://user-images.githubusercontent.com/16905302/196029302-069df83d-ad51-41fe-a367-11027b3994c1.png)

Reply:
i only see it being used for logging. and you are probably right. i guess someone just wanted to be sure we would be taken care of in the future. 😆 but i think i should leave it alone for this PR since i'm not sure the reasoning behind it and it doesn't appear to be causing any issues.